src/org/zanata/jenkins/Notifier.groovy

```diff
@@ -1,209 +1,274 @@
+            updateGitHubCommitStatus('SUCCESS', message)
+            sendEmail(message)
+    }
+
+    // Used when tests failure, but compile completed
+    void failed(String message='') {
+        sendHipChat color: "RED", notify: true, message: "FAILED: Job " + jobLinkHtml()
+            updateGitHubCommitStatus('FAILURE', message)
+            sendEmail(message)
     }
-    steps.emailext([
-      subject: "${env.JOB_NAME} - Build #${build.id} - ${build.result?:'FAILURE'}: ${message}",
-      body:  "url: ${build.absoluteUrl}\n" +
-        "      title: ${env.CHANGE_TITLE?:''}\n" +
-        "     author: ${env.CHANGE_AUTHOR}\n" +
-        "        job: ${env.JOB_NAME}\n" +
-        "   build id: ${build.id}\n" +
-        "     branch: ${env.BRANCH_NAME}\n" +
-        "     target: ${env.CHANGE_TARGET?:''}\n" +
-        "   duration: " + durationStr?:'' + " \n" +
-        "     result: ${build.result?:'FAILURE'}\n" +
-        "description: ${build.description?:''}\n" +
-        "    message: ${message}\n" +
-        "    changes: ${changes}\n",
-        recipientProviders: [
-        [$class: 'CulpritsRecipientProvider'],
-        [$class: 'RequesterRecipientProvider'],
-      ],
-    ])
-  }
-
-  private String jobLinkHtml() {
-    "<a href=\"${env.BUILD_URL}\">${env.JOB_NAME} #${env.BUILD_NUMBER}</a>"
-  }
-
-  private void sendHipChat(Map p) {
-    try {
-      steps.hipchatSend(
-              color: p.color,
-              failOnError: p.failOnError ?: false,
-              message: p.message + ( durationStr ? " Duration: " + durationStr : ''),
-              notify: p.notify ?: false,
-              sendAs: p.sendAs,
-              textFormat: p.textFormat ?: false,
-              v2enabled: p.v2enabled ?: false)
-    } catch (NoSuchMethodError ignored) {
-      steps.echo("hipchatSend skipped: no such method")
-    } catch (Exception e) {
-      // org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException is not in Maven Central,
-      // so we can't use it in Jenkins Pipeline Unit tests:
-      if (e.toString().contains('RejectedAccessException')) {
-        // allow for Jenkins In-process Script Approval
-        throw e
-      }
-      steps.echo("hipchatSend failed: " + e)
+
+    // Used when build failure. e.g. build system/script failed, or compile error
+    void error(String message='') {
+        // Need durationStr here as notify.finish might not be invoked
+        durationStr=durationToString()
+        sendHipChat color: "RED", notify: true, message: "ERROR: Job " + jobLinkHtml()
+        updateGitHubCommitStatus('ERROR', message)
+        sendEmail(message)
+    }
+
+    private String obtainCommitId(String url, String branch ) {
+        String result = null
+        String refString = null
+        if ( branch ==~ /PR-.*/ ) {
+            // Pull request does not show real branch name
+            refString = "refs/pull/" + env.CHANGE_ID + "/head"
```

Comment:
I thought the PR-xxx branch would be available in the local repo for the build. No?